src/jit/morph.cpp

```diff
@@ -83,6 +83,18 @@ GenTreePtr          Compiler::fgMorphIntoHelperCall(GenTreePtr      tree,
     tree->gtCall.gtEntryPoint.addr = nullptr;
 #endif
 
+#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+    if (varTypeIsLong(tree))
+    {
+        GenTreeCall* callNode = tree->AsCall();
+        ReturnTypeDesc* retTypeDesc = callNode->GetReturnTypeDesc();
+        retTypeDesc->Reset();
+        retTypeDesc->Initialize(this, callNode->gtRetClsHnd);
+        
+        NYI("Helper with TYP_LONG return type");
+    }
+#endif
+
```

Comment:
>  we try to decompose the parameters twice. I'm not sure why it's happening

Hmm, does the tree happen to contain a long GT_STOREIND? I discovered when I did `CreateTemporary` that the solution that was used for handling moved embedded statements can result in double lowering/decomposition.